client/src/dataset/sagas.ts

```diff
@@ -44,8 +44,16 @@ export function* doOpenDataset(fullPath: string) {
         yield put(datasetActions.Actions.detect(fullPath));
         const detectResult: DetectDatasetResponse = yield call(detectDataset, fullPath);
         if (detectResult.status === "ok") {
-            detectedParams = detectResult.datasetParams;
-            yield put(datasetActions.Actions.detected(fullPath, detectResult.datasetParams));
+            if(Object.keys(DatasetTypes).some((v) => v === detectResult.datasetParams.type)) {
+              detectedParams = detectResult.datasetParams;
+              yield put(datasetActions.Actions.detected(fullPath, detectResult.datasetParams));
+            }
+            else {
+              const timestamp = Date.now();
+              const id = uuid();
+              yield put(datasetActions.Actions.detectFailed(fullPath));
+              yield put(datasetActions.Actions.error(detectResult.datasetParams.type, detectResult.datasetParams.type + ` dataset type is currently not supported in the GUI`, timestamp, id));
```

Comment:
I think there is a small mistake here - the first argument of `Actions.error` is supposed to be the dataset uuid. So while the types may match and there is no linter or compiler error, the meaning or semantics of the value is not right.

I quickly had a look, and it seems that the dataset doesn't even have an id yet at that point. The usual sequence is:

Open File Browser -> Select file in Browser -> Send `path` to server for detection -> Show "parameters" form -> Assign UUID and "create" the dataset in the redux store.

The dataset uuid in the `error` action is currently used [to immediately remove a dataset](https://github.com/LiberTEM/LiberTEM/blob/afe3c4292a2d55d31b0b2177aff0fa5e0cc9cb51/client/src/dataset/reducers.ts#L33) for which we get this action, if it exists. As this action is idempotent, it should not hurt to just generate a new uuid in `doOpenDataset` for this purpose, which then represents the "dataset-that-should-have-been". The alternative would be  to generate the "real" uuid earlier in the sequence mentioned above, but I think this is overkill here, as it would just be removed right away anyways...

Reply:
Got it.